numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -328,28 +328,31 @@ gentype_nonzero_number(PyObject *m1)
 static PyObject *
 gentype_str(PyObject *self)
 {
-    PyObject *arr, *ret = NULL;
+    PyObject  *item, *item_str;
 
-    arr = PyArray_FromScalar(self, NULL);
-    if (arr != NULL) {
-        ret = PyObject_Str((PyObject *)arr);
-        Py_DECREF(arr);
+    item = gentype_generic_method(self, NULL, NULL, "item");
+    if (item == NULL) {
+        return NULL;
     }
-    return ret;
+
+    item_str = PyObject_Str(item);
+    Py_DECREF(item);
+    return item_str;
 }
```

Comment:
Confirmed:
```
>>> from numpy.core.test_rational import rational
>>> x = rational(1, 2); x
rational(1,2)
>>> x.item()
rational(1,2)
```

More alarmingly

```
>>> str(x)
Traceback (most recent call last):
  File "<pyshell#15>", line 1, in <module>
    str(x)
TypeError: __str__ returned non-string (type bytes)
```

Which makes it impossible to test `np.generic.__repr__(x)`